src/Microsoft.Data.SqlClient/tests/UnitTests/Handlers/LoginChainTest.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClientX.Handlers.Connection;
+using Xunit;
+
+namespace Microsoft.Data.SqlClient.NetCore.UnitTests.Handlers
+{
+    /// <summary>
+    ///  Place holder for stitching all the handlers for login and testing them
+    ///  without the orchestrator.
+    /// </summary>
+    public sealed class LoginChainTest
+    {
+
+        /// <summary>
+        ///  Datasource to use for connection test. This may be changed locally to test against a different server.
+        ///  TODO: Migrate to config when we enable CI.
+        /// </summary>
+        private string _dataSource = "tcp:localhost,1446";
```

Comment:
Is it required to be a working host or this could be any random server name/port combination?

Reply:
Expected to be random right now. The test is expecting a socket exception on this endpoint.

Reply:
I intend to evolve this test into a valid e2e test. Since we don't have the infrastructure and the login is not complete, I have expected a socket exception. But when we light up e2e then this test moves into integration tests project and this will come from config.